### DISCUSSION TITLE
cwd change manually

### DISCUSSION THREAD

**Discussion by semi710** [nvim-tree/nvim-tree.lua#2350] [Q&A] [answered]
how to make the current buffer dir as root dir on a key press ?

**Comment by alex-courtis** [comment, marked as answer]
See `:help :NvimTreeFindFile`

See `:help nvim-tree-api.tree.find_file()` for more options you can map within or outside of nvim-tree.

**Reply by semi710** [reply]
```lua

M.cwd_to_buf = function(bufnr)
	local ntok, nvimtree_api = pcall(require, "nvim-tree.api")
	if not ntok then
		vim.notify("Nvimtree call Failed!!")
		return
	end

	nvimtree_api.tree.find_file({ buf = bufnr, update_root = true, focus = true })
end

```
i Have created this fn to do that but unable to figure out how to pass bufnr to it 

as i want to change the directory to the current buffer without opening nvim-tree.

**Reply by semi710** [reply]
nvm mind was able to get it done by `:lcd %:p:h`
